src/client/pfs/pfs.proto

```diff
@@ -351,16 +350,32 @@ message AppendFileRequest {
 //  bool delete = 12;
 }
 
-message DeleteFileRequest {
+message RawFileSource {
+  string path = 1;
+  bytes data = 2;
+  bool EOF = 3;
+}
+
+message TarFileSource {
+  bytes data = 1;
+}
+
+message URLFileSource {
+  string path = 1;
+  string URL = 2;
+  bool recursive = 3;
+}
+
+message DeleteFile {
   string file = 1; 
   string tag = 2;
 }
 
-message FileOperationRequest {
+message ModifyFileRequest {
   Commit commit = 1;
```

Comment:
I think this should go inside as `Commit set_commit = 1`

Reply:
The commit and first modification message are both set in the initial modify file request message. Also, I don't think putting that in the same oneof as the modifications makes sense because they are not conceptually related.